### DISCUSSION TITLE
Base64 decode xcom

### DISCUSSION THREAD

**Discussion by a246530** [apache/airflow#21194] [Q&A] [answered]
I am using the SSH operator and writing the results to xcom ( do_xcom_push=True ). I am not using a custom xcom backend. I can see the values in the xcom table, however they are base64 encoded. I have setup a python task to decode the message but this adds another step which makes me think there has to be another way.

Does anyone have any experience with this and if so how did you handle it?

Ideally I would like to use the Jinja template but I do not see a filter that can be used ( something like | b64decode ).

**Comment by josh-fell** [comment, marked as answer]
Have you taken a look at `user_defined_macros` or `user_defined_filters` in the [Jinja templating docs](https://airflow.apache.org/docs/apache-airflow/stable/tutorial.html#templating-with-jinja)? They are pretty handy when you want to expose custom functions within Jinja templating.

**Reply by a246530** [reply]
Thank you for pointing this out. I was not fully aware of this. I just wrote a user defined filter to decode the xcom base64 value the way I initially suggested in my question and it worked just as expected.
Thanks again!
